tg_bot/modules/welcome.py

```diff
@@ -129,11 +129,15 @@ def cas_banned(userid):
 @run_async
 def new_member(bot: Bot, update: Update):
     chat = update.effective_chat  # type: Optional[Chat]
+    new_members = update.effective_message.new_chat_members
+    for new_mem in new_members:
+        if not new_mem.username:
+            chat.kick_member(new_mem.id)
```

Comment:
Are we sure that `chat` is an object on which we can go to apply the kick? Doesn't it have to be something done by the bot itself (like  `bot.restrict_chat_member()`)?

Reply:
I can check again the documentation of python-telegram-bot but it should work with chat object

Reply:
> Are we sure that `chat` is an object on which we can go to apply the kick? Doesn't it have to be something done by the bot itself (like  `bot.restrict_chat_member()`)?

https://python-telegram-bot.readthedocs.io/en/stable/telegram.chat.html?highlight=kick_member#telegram.Chat.kick_member
It's a shortcut of bot.kick_member
What we should check is the Note section.